libr/asm/p/asm_x86_nz.c

```diff
@@ -4913,6 +4910,14 @@ static int parseOperand(RAsm *a, const char *str, Operand *op, bool isrepop) {
 			str = ++p;
 		}
 		op->immediate = getnum (a, str);
+		if (op->immediate == -1) {
+			op->is_good_flag = false;
+			char *num_str = r_str_trim_dup (str);
+			if (num_str) {
+				op->is_good_flag = !r_str_casecmp (num_str, "0xffffffffffffffff");
```

Comment:
i still dont see why are you comparing the string, if the sole reason is to differenciate -1 of error in getnum with UT64_MAX, just check the error bit in RNum. comparing 0xfffff seems wrong to me

Reply:
>  comparing 0xfffff seems wrong to me

it feels wrong to me too but technically not a blocker and the offending code is small ... and depending on RNum's error bit means depending on RNum's error handling code